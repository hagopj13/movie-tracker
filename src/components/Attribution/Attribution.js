// @flow
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

import tmdbLogo from './tmdbLogo.svg';

const useStyles = makeStyles((theme) => ({
  root: {
    padding: theme.spacing(2, 3),
    display: 'flex',
    alignItems: 'center',
    minWidth: 270,
    [theme.breakpoints.down('xs')]: {
      flexDirection: 'column',
      maxWidth: 330,
    },
  },
  logo: {
    width: 110,
    height: 52,
    marginRight: theme.spacing(3),
    [theme.breakpoints.down('xs')]: {
      marginRight: 0,
      marginBottom: theme.spacing(2),
    },
  },
  text: {
    fontWeight: theme.typography.fontWeightRegular,
    [theme.breakpoints.down('xs')]: {
      textAlign: 'center',
    },
  },
}));

const Attribution = () => {
  const classes = useStyles();

  return (
    <Paper className={classes.root}>
      <img className={classes.logo} src={tmdbLogo} alt="tmdb logo" />
      <Typography className={classes.text} variant="h6">
        This product uses the TMDb API but is not endorsed or certified by TMDb.
      </Typography>
    </Paper>
  );
};

export default Attribution;
