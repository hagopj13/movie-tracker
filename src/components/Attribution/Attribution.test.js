import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import Attribution from './Attribution';

describe('Attribution component', () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow();
  });

  it('should render the Attribution component correctly', () => {
    const wrapper = shallow(<Attribution />);
    expect(wrapper).toMatchSnapshot();
  });
});
