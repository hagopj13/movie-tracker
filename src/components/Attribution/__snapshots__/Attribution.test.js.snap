// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attribution component should render the Attribution component correctly 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-root-1"
>
  <img
    alt="tmdb logo"
    className="makeStyles-logo-2"
    src="tmdbLogo.svg"
  />
  <WithStyles(ForwardRef(Typography))
    className="makeStyles-text-3"
    variant="h6"
  >
    This product uses the TMDb API but is not endorsed or certified by TMDb.
  </WithStyles(ForwardRef(Typography))>
</WithStyles(ForwardRef(Paper))>
`;
